pkg/services/alerting/notifiers/slack.go

```diff
@@ -291,15 +291,17 @@ func (sn *SlackNotifier) Notify(evalContext *alerting.EvalContext) error {
 		attachment["image_url"] = imageURL
 	}
 	body := map[string]interface{}{
-		"text":   evalContext.GetNotificationTitle(),
-		"blocks": blocks,
+		"text": evalContext.GetNotificationTitle(),
 		"attachments": []map[string]interface{}{
 			attachment,
 		},
 		"parse": "full", // to linkify urls, users and channels in alert message.
 	}
 
 	//recipient override
+	if len(blocks) > 0 {
```

Comment:
What is the practical effect of only including blocks if non-empty?

Reply:
Also, how is this change relevant to the PR's topic?

Reply:
Without this change, `json.Marshal()` output `"blocks":null,` to POST body for `chat.postMessage` .
I'm not sure Slack ignore this noisy parameter, but it is safe to remove.

Reply:
The comment `//recipient override` is also now in the wrong place, since it applies to line 305.

Reply:
Sorry, I didn't care about comment. I move and simplify the code. Would you check?